src/modules/acbs/acbs-facility-loan.service.test.ts

```diff
@@ -0,0 +1,149 @@
+          of({
+            data: facilityLoansInAcbs,
+            status: 200,
+            statusText: 'Ok',
+            config: undefined,
+            headers: undefined,
+          }),
+        );
+
+      const loans = await service.getLoansForFacility(portfolioIdentifier, facilityIdentifier, idToken);
+
+      expect(loans).toBe(facilityLoansInAcbs);
+    });
+
+    it('returns an empty array if ACBS responds with an empty array', async () => {
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArgs)
+        .mockReturnValueOnce(
+          of({
+            data: [],
+            status: 200,
+            statusText: 'Ok',
+            config: undefined,
+            headers: undefined,
+          }),
+        );
+
+      const loans = await service.getLoansForFacility(portfolioIdentifier, facilityIdentifier, idToken);
+
+      expect(loans).toStrictEqual([]);
+    });
+
+    it('throws an AcbsResourceNotFoundException if ACBS responds with a 400 response that is a string containing "Facility not found"', async () => {
+      const axiosError = new AxiosError();
+      axiosError.response = {
+        data: 'Facility not found or user does not have access to it.',
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getLoansPromise = service.getLoansForFacility(portfolioIdentifier, facilityIdentifier, idToken);
+
+      await expect(getLoansPromise).rejects.toBeInstanceOf(AcbsResourceNotFoundException);
+      await expect(getLoansPromise).rejects.toThrow(`Facility with identifier ${facilityIdentifier} was not found by ACBS.`);
+      await expect(getLoansPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws an AcbsException if ACBS responds with a 400 response that is a string that does NOT contain "The facility not found"', async () => {
+      const axiosError = new AxiosError();
+      axiosError.response = {
+        data: 'some error string',
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getLoansPromise = service.getLoansForFacility(portfolioIdentifier, facilityIdentifier, idToken);
+
+      await expect(getLoansPromise).rejects.toBeInstanceOf(AcbsException);
+      await expect(getLoansPromise).rejects.toThrow(`Failed to get the loans for the facility with identifier ${facilityIdentifier}.`);
+      await expect(getLoansPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws an AcbsException if ACBS responds with a 400 response that is NOT a string', async () => {
```

Comment:
Can we add few additional assertions please?
1. Invalid facility ID
2. Invalid portfolio ID
3. 404 when facility loan does not exists.

Reply:
1. The check for facility ID validity happens at the controller (not the service) level, so I don't think it makes sense to have an assertion about that here. As this check is common to several endpoints, it's a TODO to add a reusable API test for this (https://ukef-dtfs.atlassian.net/browse/APIM-202).
2. This isn't handled at the endpoint level (as the portfolio ID is included in the base URL).
3. If there are no loans for the facility, we return a 200 empty array. If you mean the case when the facility does not exist, this is covered by the assertion `throws an AcbsResourceNotFoundException if ACBS responds with a 400 response that is a string containing "Facility not found"`. The `AcbsResourceNotFoundException` gets mapped to a 404 by the interceptor.